Makefile-tests.am

```diff
@@ -25,7 +25,10 @@ include $(top_srcdir)/buildutil/glib-tap.mk
 TESTS_ENVIRONMENT += OT_TESTS_DEBUG=1 \
 	SRCDIR=$$(cd $(top_srcdir) && pwd) \
 	BUILDDIR=$$(cd $(top_builddir) && pwd) \
-	PATH=$$(cd $(top_builddir) && pwd):$${PATH}
+	GI_TYPELIB_PATH=$$(cd $(top_builddir) && pwd) \
```

Comment:
FWIW this sort of stuff is exactly the reason why I think InstalledTests is a generally more robust system.  But it is hard to beat the convenience and historical momentum behind `make check`, so we'll keep doing both.

It might be nice if there were a cargo-cultable `.am` fragment we could share for this?  Anyways this is obviously fine as is.